src/SmallStrainPlasticity/SmallStrainPlasticity.jl

```diff
@@ -0,0 +1,291 @@
+
+Von Mises yield function:
+```math
+\\Phi = \\sqrt{\\frac{3}{2}} \\left| \\text{dev} \\left( \\boldsymbol{\\sigma} - \\boldsymbol{\\beta} \\right) \\right| - \\sigma_y - \\kappa
+```
+where ``\\boldsymbol{\\beta} = \\sum_{i=1}^{N_\\mathrm{kin} \\boldsymbol{\\beta}_i`` is the total back-stress. 
+The evolution of ``\\boldsymbol{\\beta}_i`` is given by the kinematic hardening, specified below and using `m.kinematic`
+
+Associative plastic flow is used to obtain the plastic strains,
+```math
+\\dot{\\epsilon}_{\\mathrm{p}} = \\dot{\\lambda} \\frac{\\partial \\Phi}{\\boldsymbol{\\sigma}}
+= \\dot{\\lambda} \\boldsymbol{\\nu}
+```
+
+The plastic multiplier, ``\\lambda``, is obtained via the KKT-conditions,
+```math
+\\dot{\\lambda} \\geq 0, \\quad \\Phi \\leq 0, \\quad \\dot{\\lambda}\\Phi = 0
+```
+
+The isotropic hardening is formulated as
+```math
+\\kappa = \\sum_{i=1}^{N_{\\mathrm{iso}}} g_{\\mathrm{iso},i}(\\lambda)
+```
+where ``g_{\\mathrm{iso},i}(\\lambda)`` is specified by `m.isotropic[i]`
+
+Kinematic hardening is formulated as
+```math
+\\boldsymbol{\\beta}_i = \\dot{\\lambda} g_{\\mathrm{kin},i}(\\nu, \\boldsymbol{\\beta}_i)
+```
+where ``g_{\\mathrm{kin},i}(\\boldsymbol{\\nu}, \\boldsymbol{\\beta}_i)`` is specified by `m.kinematic[i]`
+and ``i\\in[1,N_\\mathrm{kin}]``. 
+
+```
+# Keyword arguments
+- `cache`: Cache for the iterative solver, used by NLsolve.jl. It is strongly recommended to pre-allocate the cache for repeated calls to `material_response`. See [`get_cache`](@ref).
+- `options::Dict{Symbol, Any}`: Solver options for the non-linear solver. Under the key `:nlsolve_params` keyword arguments for `nlsolve` can be handed over.
+See [NLsolve documentation](https://github.com/JuliaNLSolvers/NLsolve.jl#common-options). By default the Newton solver will be used.
+"""
+function material_response(material::Chaboche, ϵ::SymmetricTensor{2,3}, state_old::ChabocheState{Nkin,T,N}, Δt; cache=get_cache(material), options::Dict{Symbol, Any} = Dict{Symbol, Any}()) where {T,N,Nkin}
+    
+    σ_trial, dσdϵ_elastic, _, _ = material_response(material.elastic, ϵ-state_old.ϵₚ, nothing, Δt)
+
+    Φ_trial = yieldCriterion(material, σ_trial-sum(state_old.β), state_old.λ)
+    if Φ_trial < 0
+        return σ_trial, dσdϵ_elastic, state_old, true
+    else
+        num_blas_threads = LinearAlgebra.BLAS.get_num_threads() # (~ 2ns)
+        LinearAlgebra.BLAS.set_num_threads(1)                   # Big performance benefit, takes ~8ns
+        converged = solve_local_problem!(cache, material, state_old, ϵ, options)
+        if converged
+            σ, dσdϵ, state = get_plastic_output(cache, material, state_old, σ_trial, dσdϵ_elastic)
+        else
+            σ, dσdϵ, state = (σ_trial, dσdϵ_elastic, state_old)
+            println("Did not converge!")
+        end
+        LinearAlgebra.BLAS.set_num_threads(num_blas_threads)
+        return σ, dσdϵ, state, converged
+    end
+    
+end
+
+include("Residual.jl")
```

Comment:
Maybe skip a separate residual file and paste it here instead?

Reply:
True!